### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by gaoteng-hi** [kuz/DeepMind-Atari-Deep-Q-Learner#11]
Excuse me. I installed it in "Ubuntu 14.04.3 LTS". When I run "./run_cpu breakout", it shows the following error. Would you please help me on this? Thanks a lot in advance!
# ./run_cpu breakout

-framework alewrap -game_path /home/gaoteng/DeepmindAtari/DeepMind-Atari-Deep-Q-Learner-master/roms/ -name DQN3_0_1_breakout_FULL_Y -env breakout -env_params useRGB=true -agent NeuralQLearner -agent_params lr=0.00025,ep=1,ep_end=0.1,ep_endt=replay_memory,discount=0.99,hist_len=4,learn_start=50000,replay_memory=1000000,update_freq=4,n_replay=1,network="convnet_atari3",preproc="net_downsample_2x_full_y",state_dim=7056,minibatch_size=32,rescale_r=1,ncols=1,bufferSize=512,valid_size=500,target_q=10000,clip_delta=1,min_reward=-1,max_reward=1 -steps 50000000 -eval_freq 250000 -eval_steps 125000 -prog_freq 5000 -save_freq 125000 -actrep 4 -gpu -1 -random_starts 30 -pool_frms type="max",size=2 -seed 1 -threads 4
Torch Threads:  4
Using CPU code only. GPU device id: -1
Torch Seed: 1

Playing:    breakout
Creating Agent Network from convnet_atari3
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> (11) -> output](1): nn.Reshape(4x84x84)
  (2): nn.SpatialConvolution(4 -> 32, 8x8, 4,4, 1,1)
  (3): nn.Rectifier
  (4): nn.SpatialConvolution(32 -> 64, 4x4, 2,2)
  (5): nn.Rectifier
  (6): nn.SpatialConvolution(64 -> 64, 3x3)
  (7): nn.Rectifier
  (8): nn.Reshape(3136)
  (9): nn.Linear(3136 -> 512)
  (10): nn.Rectifier
  (11): nn.Linear(512 -> 4)
}
Convolutional layers flattened output size: 3136
./run_cpu: line 46:  2372 Segmentation fault      (core dumped) ../torch/bin/qlua train_agent.lua $args

**Comment by Ashram56** [comment]
Good morning,

I have exactly the same issue, both GPU and CPU version, when trying to run on a Jetson TX1.

Anyone would have a solution for this issue ?

Regards

**Comment by gaoteng-git** [comment]
This problem is caused by lack of memory. It consumes 6GB+ memory on start. If not enough on the machine, it crashes like this.

**Comment by J6767** [comment]
I have the same problem. Pacman and Breakout run, but I get the seg fault for spaceinvaders and montzrev. I have a TitanX, so do these require more than 12GB?

**Comment by jkjung-avt** [comment]
I can get around this problem by just reducing the size of replay_memory. I've created a blog post and forked a GitHub repository for it. Feel free to check it out and provide comments.

[Training DeepMind's DQN to Play 'Pong', the Atari Game](https://jkjung-avt.github.io/dqn-pong/)
[https://github.com/jkjung-avt/DeepMind-Atari-Deep-Q-Learner](https://github.com/jkjung-avt/DeepMind-Atari-Deep-Q-Learner)
